src/backend/distributed/utils/attribute.c

```diff
@@ -600,3 +575,63 @@ MultiTenantMonitorshmemSize(void)
 
 	return size;
 }
+
+
+/*
+ * extractTopComment extracts the top-level multi-line comment from a given input string.
+ */
+static char *
+extractTopComment(const char *inputString)
+{
+	int commentStartCharsLength = 2;
+	if (strlen(inputString) < commentStartCharsLength)
+	{
+		return NULL;
+	}
+
+	int i = 0;
+
+	/* If query starts with a comment */
+	if (inputString[i] == '/' && inputString[i + 1] == '*')
+	{
+		/* Skip the comment start characters */
+		i += 2;
+		while (inputString[i] && (inputString[i] != '*' && inputString[i + 1] != '/'))
```

Comment:
do we plan to do some kind of escaping to handle situations like tenant_id = `"akdhek*/a"`?